api/maven-api-core/src/main/java/org/apache/maven/api/Session.java

```diff
@@ -55,23 +56,32 @@ public interface Session {
     SessionData getData();
 
     /**
-     * Gets the user properties to use for interpolation. The user properties have been configured directly by the user,
-     * e.g. via the {@code -Dkey=value} parameter on the command line.
+     * Returns immutable user properties to use for interpolation. The user properties have been configured directly
+     * by the user, e.g. via the {@code -Dkey=value} parameter on the command line.
      *
      * @return the user properties, never {@code null}
      */
     @Nonnull
     Map<String, String> getUserProperties();
 
     /**
-     * Gets the system properties to use for interpolation. The system properties are collected from the runtime
-     * environment such as {@link System#getProperties()} and environment variables.
+     * Returns immutable system properties to use for interpolation. The system properties are collected from the
+     * runtime environment such as {@link System#getProperties()} and environment variables.
      *
      * @return the system properties, never {@code null}
      */
     @Nonnull
     Map<String, String> getSystemProperties();
 
+    /**
+     * Each invocation computes a new map of effective properties. To be used in interpolation.
```

Comment:
I think we need to explain here what "effective properties" are.

Reply:
Done